server/src/main/java/org/elasticsearch/index/shard/IndexShardOperationPermits.java

```diff
@@ -123,23 +123,66 @@ public void close() {
      * @param onFailure the action to run if a failure occurs while blocking operations
      * @param <E>       the type of checked exception thrown by {@code onBlocked} (not thrown on the calling thread)
      */
-    <E extends Exception> void asyncBlockOperations(
-            final long timeout, final TimeUnit timeUnit, final CheckedRunnable<E> onBlocked, final Consumer<Exception> onFailure) {
+    <E extends Exception> void asyncBlockOperations(final long timeout, final TimeUnit timeUnit,
```

Comment:
since we only have one user of this api, I wonder wether we should just remove this syntactic sugar method and put the single user to the other api